nilearn/maskers/base_masker.py

```diff
@@ -424,3 +429,105 @@ def __sklearn_tags__(self):
             surf_img=True, niimg_like=False, masker=True
         )
         return tags
+
+    def transform(self, imgs, confounds=None, sample_mask=None):
+        """Apply mask, spatial and temporal preprocessing.
+
+        Parameters
+        ----------
+        imgs : 1D/2D Surface image
+            Images to process.
+
+        confounds : CSV file or array-like, default=None
+            This parameter is passed to signal.clean. Please see the related
+            documentation for details.
+            shape: (number of scans, number of confounds)
+
+        sample_mask : Any type compatible with numpy-array indexing, \
+            default=None
+            shape: (number of scans - number of volumes removed, )
+            Masks the niimgs along time/fourth dimension to perform scrubbing
+            (remove volumes with high motion) and/or non-steady-state volumes.
+            This parameter is passed to signal.clean.
+
+        Returns
+        -------
+        region_signals : 2D numpy.ndarray
+            Signal for each element.
+            shape: (number of scans, number of elements)
+
+        """
+        check_is_fitted(self)
+
+        if not isinstance(imgs, list):
+            imgs = [imgs]
+        imgs = concat_imgs(imgs)
+
+        check_compatibility_mask_and_images(self.mask_img_, imgs)
+
+        if self.mask_img_ is not None:
+            check_same_n_vertices(self.mask_img_.mesh, imgs.mesh)
+
+        if self.smoothing_fwhm is not None:
+            warnings.warn(
+                "Parameter smoothing_fwhm "
+                "is not yet supported for surface data",
+                UserWarning,
+                stacklevel=find_stack_level(),
+            )
+            self.smoothing_fwhm = None
+
+        if self.reports:
+            self._reporting_data["images"] = imgs
+
+        if confounds is None and not self.high_variance_confounds:
+            return self.transform_single_imgs(
+                imgs, confounds=confounds, sample_mask=sample_mask
+            )
+
+        # Compute high variance confounds if requested
+        all_confounds = []
+
+        if self.high_variance_confounds:
+            hv_confounds = self._cache(high_variance_confounds)(imgs)
+            all_confounds.append(hv_confounds)
```

Comment:
It should be possible to cover these lines ?

Reply:
added a new estimator check that should run on all maskers